tournaments/server_requests.py

```diff
@@ -0,0 +1,20 @@
+from .models import Tournament
+from edagames import settings
+import requests
+from typing import List
+from itertools import combinations
+
+
+def generate_combination(
+    tournament_name: str,
+    bot_list: List[str],
+):
+    tournament_id = Tournament.objects.only('id').get(name=tournament_name).id
+    data = {
+        'tournament_id': str(tournament_id),
+        'challenges': list(combinations(bot_list.split(sep=','), 2)),
+    }
+    return requests.post(
+        f'{settings.SERVER_URL}:{settings.SERVER_PORT}/tournament',
+        json=data,
+    )
```

Comment:
This method has 2 responsibilities.

1. Create the match combinations.
2. Send the data to Server.

You should move the combination to another method. If not, how are you sure that the list is what you expect?

Reply:
but combinations is something that is done by itertools